app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -0,0 +1,430 @@
+  }
+
+  private fun getScreenCentreY(): Int {
+    Log.d("overlay screenCentre Y", (screenHeight / 2).toString())
+    return screenHeight / 2
+  }
+
+  private fun getScreenCentreX(): Int {
+    Log.d("overlay screenCentre X", (screenWidth / 2).toString())
+    return screenWidth / 2
+  }
+
+  private fun calculateAnchorPosition(spotlightTarget: SpotlightTarget): AnchorPosition {
+    Log.d("overlay", "checking anchor position. RTL = $isRtl")
+    return if (spotlightTarget.anchorCentreX > getScreenCentreX()) {
+      if (spotlightTarget.anchorCentreY > getScreenCentreY()) {
+        AnchorPosition.BottomRight
+      } else {
+        AnchorPosition.TopRight
+      }
+    } else if (spotlightTarget.anchorCentreY > getScreenCentreY()) {
+      AnchorPosition.BottomLeft
+    } else {
+      AnchorPosition.TopLeft
+    }
+  }
+
+  private fun requestOverlayResource(spotlightTarget: SpotlightTarget): View {
+    val anchorPosition = calculateAnchorPosition(spotlightTarget)
+    Log.d("overlay", anchorPosition.toString())
+
+    return when (anchorPosition) {
+      AnchorPosition.TopLeft -> {
+        if (isRtl) {
+          configureTopRightOverlay(spotlightTarget)
+        } else {
+          configureTopLeftOverlay(spotlightTarget)
+        }
+      }
+      AnchorPosition.TopRight -> {
+        if (isRtl) {
+          configureTopLeftOverlay(spotlightTarget)
+        } else {
+          configureTopRightOverlay(spotlightTarget)
+        }
+      }
+      AnchorPosition.BottomRight -> {
+        if (isRtl) {
+          configureBottomLeftOverlay(spotlightTarget)
+        } else {
+          configureBottomRightOverlay(spotlightTarget)
+        }
+      }
+      AnchorPosition.BottomLeft -> {
+        if (isRtl) {
+          configureBottomRightOverlay(spotlightTarget)
+        } else {
+          configureBottomLeftOverlay(spotlightTarget)
+        }
+      }
+    }
+  }
+
+  private sealed class AnchorPosition {
+    /** The position corresponding to the anchor when it is on the top left of the screen. */
+    object TopLeft : AnchorPosition()
+    /** The position corresponding to the anchor when it is on the top right of the screen. */
+    object TopRight : AnchorPosition()
+    /** The position corresponding to the anchor when it is on the bottom left of the screen. */
+    object BottomLeft : AnchorPosition()
+    /** The position corresponding to the anchor when it is on the bottom right of the screen. */
+    object BottomRight : AnchorPosition()
+  }
+
+  override fun clickOnNextTip() {
+    spotlight.next()
+  }
+
+  private fun configureBottomLeftOverlay(spotlightTarget: SpotlightTarget): View {
+    overlayBinding = BottomLeftOverlayBinding.inflate(this.layoutInflater)
+    (overlayBinding as BottomLeftOverlayBinding).let {
+      it.lifecycleOwner = this
+      it.listener = this
+    }
+
+    (overlayBinding as BottomLeftOverlayBinding).customText.text = spotlightTarget.hint
+
```

Comment:
Nit: remove this & the above extra newlines.

Reply:
done